library/kani/src/arbitrary.rs

```diff
@@ -1,33 +1,103 @@
 // Copyright Kani Contributors
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-//! This module introduces the Arbitrary trait as well as implementation for the Invariant trait.
-use crate::{any_raw, assume, Invariant};
+//! This module introduces the Arbitrary trait as well as implementation for primitive types and
+//! other std containers.
+use std::num::*;
 
 /// This trait should be used to generate symbolic variables that represent any valid value of
 /// its type.
 pub trait Arbitrary {
     fn any() -> Self;
 }
 
-impl<T> Arbitrary for T
-where
-    T: Invariant,
-{
-    default fn any() -> Self {
-        let value = unsafe { any_raw::<T>() };
-        assume(value.is_valid());
-        value
+/// The given type can be represented by an unconstrained symbolic value of size_of::<T>.
+macro_rules! trivial_arbitrary {
+    ( $type: ty ) => {
+        impl Arbitrary for $type {
+            #[inline(always)]
+            fn any() -> Self {
+                unsafe { crate::any_raw_internal::<$type>() }
+            }
+        }
+    };
+}
+
+trivial_arbitrary!(u8);
+trivial_arbitrary!(u16);
+trivial_arbitrary!(u32);
+trivial_arbitrary!(u64);
+trivial_arbitrary!(u128);
+trivial_arbitrary!(usize);
+
+trivial_arbitrary!(i8);
+trivial_arbitrary!(i16);
+trivial_arbitrary!(i32);
+trivial_arbitrary!(i64);
+trivial_arbitrary!(i128);
+trivial_arbitrary!(isize);
+
+// We do not constraint floating points values per type spec. Users must add assumptions to their
+// verification code if they want to eliminate NaN, infinite, or subnormal.
+trivial_arbitrary!(f32);
+trivial_arbitrary!(f64);
+
+trivial_arbitrary!(());
```

Comment:
What would this show up as in the CBMC output?

Reply:
That's a great question! We actually don't generate this value in the compiler. https://github.com/model-checking/kani/blob/4465b9819286bea550f3f62fdf6322a55939500f/kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs#L206-L208